### PR TITLE
Fix for #11719 - TSServer: JS files should display syntactic errors for TS syntax  

### PR DESCRIPTION
**Pull request by jramsay**
Fix for #11719 - TSServer: JS files should display syntactic errors for TS syntax 

For JavaScript files, report semantic errors for using TypeScript-only constructs from within a JavaScript file as syntactic errors.

### CODE REVIEW

**Comment by jramsay** [comment]
// cc: @mhegazy, @vladima , @billti

File: src/compiler/program.ts [hunk 1]
```diff
@@ -725,6 +725,12 @@ namespace ts {
         }
 
         function getSyntacticDiagnosticsForFile(sourceFile: SourceFile): Diagnostic[] {
+            // For JavaScript files, we report semantic errors for using TypeScript-only
+            // constructs from within a JavaScript file as syntactic errors.
+            if (isSourceFileJavaScript(sourceFile) && !sourceFile.parseJavaScriptDiagnostics) {
+                sourceFile.parseJavaScriptDiagnostics = getJavaScriptSemanticDiagnosticsForFile(sourceFile);
```
Comment (mhegazy) [inline src/compiler/program.ts]: change the name to `getJavaScriptSyntacticDiagnosticsForFile`

File: src/compiler/program.ts [hunk 2]
```diff
@@ -725,6 +725,12 @@ namespace ts {
         }
 
         function getSyntacticDiagnosticsForFile(sourceFile: SourceFile): Diagnostic[] {
+            // For JavaScript files, we report semantic errors for using TypeScript-only
+            // constructs from within a JavaScript file as syntactic errors.
+            if (isSourceFileJavaScript(sourceFile) && !sourceFile.parseJavaScriptDiagnostics) {
+                sourceFile.parseJavaScriptDiagnostics = getJavaScriptSemanticDiagnosticsForFile(sourceFile);
+                sourceFile.parseDiagnostics = sourceFile.parseDiagnostics.concat(sourceFile.parseJavaScriptDiagnostics);
```
Comment (mhegazy) [inline src/compiler/program.ts]: i would not change the parseDiagnostics here.

File: src/compiler/program.ts [hunk 3]
```diff
@@ -725,6 +725,12 @@ namespace ts {
         }
 
         function getSyntacticDiagnosticsForFile(sourceFile: SourceFile): Diagnostic[] {
+            // For JavaScript files, we report semantic errors for using TypeScript-only
+            // constructs from within a JavaScript file as syntactic errors.
+            if (isSourceFileJavaScript(sourceFile) && !sourceFile.parseJavaScriptDiagnostics) {
+                sourceFile.parseJavaScriptDiagnostics = getJavaScriptSemanticDiagnosticsForFile(sourceFile);
+                sourceFile.parseDiagnostics = sourceFile.parseDiagnostics.concat(sourceFile.parseJavaScriptDiagnostics);
+            }
```
Comment (mhegazy) [inline src/compiler/program.ts]: consider changing it to:

```ts
   if (isSourceFileJavaScript(sourceFile))
            if (!sourceFile.additionalSyntacticDiagnostics) {
                sourceFile.additionalSyntacticDiagnostics = getJavaScriptAdditionalSyntacticDiagnosticsForFile(sourceFile);
            }
        return concatenate(sourceFile.additionalSyntacticDiagnostics, sourceFile.parseDiagnostics);
    }
    return sourceFile.parseDiagnostics;
```

File: src/compiler/program.ts (hunk 1 above)
Comment (jramsay) [inline src/compiler/program.ts]: done

File: src/compiler/program.ts (hunk 3 above)
Comment (jramsay) [inline src/compiler/program.ts]: sounds good - done

**Pull request merged**